lib/eeny-meeny/railtie.rb

```diff
@@ -15,8 +15,20 @@ class Railtie < Rails::Railtie
         config.secure                = app.config.eeny_meeny[:secure]           if app.config.eeny_meeny.key?(:secure)
         config.query_parameters      = app.config.eeny_meeny[:query_parameters] if app.config.eeny_meeny.key?(:query_parameters)
       end
-      # Insert Middleware
-      app.middleware.insert_before ActionDispatch::Cookies, EenyMeeny::Middleware
+
+      experiment_with_dependency = EenyMeeny::Experiment.find_all.detect { |experiment| experiment.smoke_test_dependency.present? }
+
+      if experiment_with_dependency
+        # Insert middleware after the user session is available so that the host can write logic
+        # dependent on the user session. We only do so of we have experiments which depend on
+        # a smoke test as this intrduces a pottential limitation of users not being bucketed
+        # on the first request and the cookie will be available in the controller only
+        # after a page refresh.
+        app.middleware.insert_after ActionDispatch::Session::CookieStore, EenyMeeny::Middleware
+      else
+        # Insert middleware
+        app.middleware.insert_before ActionDispatch::Cookies, EenyMeeny::Middleware
+      end
```

Comment:
Do we really need to have this conditionals of middleware ordering? Such different ordering may introduce some rabbit holes, so I wonder if we can simply do `insert_after` always.

Reply:
Well this is something I'd like to discuss with @corthmann as well. I like this approach as it only impacts the case when we need to use conditional experiments. Also, the middleware used on the host (i.e. ruby-web) would be used as complementary tool to smoke test dependent experiments and should only be configured when we have such an experiment.

Anyway, this should hopefully be a one-off thing for the CRO2.